### ISSUE TITLE
Cannot load CSV.jl at all (precompilation)

### ISSUE THREAD

**Issue by mileslucas** [JuliaData/CSV.jl#976]
I can't use (seemingly) any version of CSV.jl on my current set up. I am using Mac OS X Monterey with a Julia version built for x86

```julia
julia> versioninfo()
Julia Version 1.7.1
Commit ac5cc99908 (2021-12-22 19:35 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin19.5.0)
  CPU: Apple M1 Max
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-12.0.1 (ORCJIT, westmere)
Environment:
  JULIA_NUM_THREADS = auto
```

Whenever I 
```julia
using CSV
```
I get stuck precompiling and it never ends. Like, I ran it overnight without success. `ctrl-c` gives "Interrupted: Exiting precompilation...", so no useful information there.

I've tried reverting to `0.9` versions, no success, I've tried explicitly trying `0.10.0` `0.10.1` and `0.10.2` and none of them work. I've tried checking out from `main` without success.

**Comment by ufechner7** [comment]
What is the output of the command:
```
]st
```
?

**Comment by mileslucas** [comment]
It doesn't work in a clean environment-

```julia
(@v1.7) pkg> activate --temp
  Activating new project at `/var/folders/q6/j_0w4_mx13jg_7mw9__902f40000gn/T/jl_P39Qal`

(jl_P39Qal) pkg> add CSV
    Updating registry at `~/.julia/registries/General.toml`
   Resolving package versions...
    Updating `/private/var/folders/q6/j_0w4_mx13jg_7mw9__902f40000gn/T/jl_P39Qal/Project.toml`
  [336ed68f] + CSV v0.10.2
    Updating `/private/var/folders/q6/j_0w4_mx13jg_7mw9__902f40000gn/T/jl_P39Qal/Manifest.toml`
  [336ed68f] + CSV v0.10.2
  [944b1d66] + CodecZlib v0.7.0
  [34da2185] + Compat v3.41.0
  [9a962f9c] + DataAPI v1.9.0
  [e2d170a0] + DataValueInterfaces v1.0.0
  [48062228] + FilePathsBase v0.9.17
  [842dd82b] + InlineStrings v1.1.2
  [82899510] + IteratorInterfaceExtensions v1.0.0
  [69de0a69] + Parsers v2.2.1
  [2dfb63ee] + PooledArrays v1.4.0
  [91c51154] + SentinelArrays v1.3.11
  [3783bdb8] + TableTraits v1.0.1
  [bd369af6] + Tables v1.6.1
  [3bb67fe8] + TranscodingStreams v0.9.6
  [ea10d353] + WeakRefStrings v1.4.1
  [0dad84c5] + ArgTools
  [56f22d72] + Artifacts
  [2a0f44e3] + Base64
  [ade2ca70] + Dates
  [8bb1440f] + DelimitedFiles
  [8ba89e20] + Distributed
  [f43a241f] + Downloads
  [9fa8497b] + Future
  [b77e0a4c] + InteractiveUtils
  [b27032c2] + LibCURL
  [76f85450] + LibGit2
  [8f399da3] + Libdl
  [37e2e46d] + LinearAlgebra
  [56ddb016] + Logging
  [d6f4376e] + Markdown
  [a63ad114] + Mmap
  [ca575930] + NetworkOptions
  [44cfe95a] + Pkg
  [de0858da] + Printf
  [3fa0cd96] + REPL
  [9a3f8284] + Random
  [ea8e919c] + SHA
  [9e88b42a] + Serialization
  [1a1011a3] + SharedArrays
  [6462fe0b] + Sockets
  [2f01184e] + SparseArrays
  [10745b16] + Statistics
  [fa267f1f] + TOML
  [a4e569a6] + Tar
  [8dfed614] + Test
  [cf7118a7] + UUIDs
  [4ec0a83e] + Unicode
  [e66e0078] + CompilerSupportLibraries_jll
  [deac9b47] + LibCURL_jll
  [29816b5a] + LibSSH2_jll
  [c8ffd9c3] + MbedTLS_jll
  [14a3606d] + MozillaCACerts_jll
  [4536629a] + OpenBLAS_jll
  [83775a58] + Zlib_jll
  [8e850b90] + libblastrampoline_jll
  [8e850ede] + nghttp2_jll
  [3f19e933] + p7zip_jll
Precompiling project...
  Progress [>                                        ]  0/1
  ◒ CSV
^C Interrupted: Exiting precompilation...

(jl_P39Qal) pkg> st
      Status `/private/var/folders/q6/j_0w4_mx13jg_7mw9__902f40000gn/T/jl_P39Qal/Project.toml`
  [336ed68f] CSV v0.10.2
```

**Comment by quinnj** [comment]
Note that environments "stack", so even if you make a "new" environment, packages may be pulled from "parent" environments. There's also a way to turn off automatic precompilation, though I think when you first call `using CSV`, it will still require precompiling before the package will load.

Otherwise, I would try commenting out the precompilation that CSV.jl does (in [this file](https://github.com/JuliaData/CSV.jl/blob/main/src/precompile.jl)) and see if that fixes it? Maybe something there is hanging? Definitely weird because it seems like it's something specific to your OS/machine.

**Comment by mileslucas** [comment]
Okay, in a local clone, if I remove the `_precompile_()` function definition (I just replaced it with `_precomiple_() = nothing`) Everything works fine. Without that function changed, I can't `using CSV` (even in a dockerized, fresh, CSV.jl only install).

The actual line that hangs during precompilation (by trial and error), is
```julia
CSV.Context(joinpath(dirname(pathof(CSV)), "promotions.csv"))
```

**Comment by mileslucas** [comment]
Also, as a note, this doesn't exist for my version of julia built on master for `aarch64`. So this may be some weird rosetta-only bug; I'm happy to keep digging!

**Comment by adigitoleo** [comment]
FWIW I also can't precompile anymore, on Linux, some long method error from Parsers.Options

<details><summary>Stack trace</summary>

```
julia> versioninfo()
Julia Version 1.7.1
Commit ac5cc99908 (2021-12-22 19:35 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-8705G CPU @ 3.10GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-12.0.1 (ORCJIT, skylake)

(@v1.7) pkg> st CSV
      Status `~/.julia/environments/v1.7/Project.toml`
  [336ed68f] CSV v0.10.0

julia> using CSV
[ Info: Precompiling CSV [336ed68f-0bac-5ca0-87d4-7b16caf5d00b]
ERROR: LoadError: MethodError: no method matching Parsers.Options(::Missing, ::UInt8, ::UInt8, ::UInt8, ::UInt8, ::UInt8, ::UInt8, ::UInt8, ::Vector{String}, ::Vector{String}, ::Nothing, ::Bool, ::Bool, ::Nothing, ::Bool, ::Bool, ::Bool)
Closest candidates are:
  Parsers.Options(::Union{Missing, Nothing, Vector{String}}, ::Union{Char, UInt8}, ::Union{Char, UInt8}, ::Union{Char, UInt8}, ::Union{Char, UInt8}, ::Union{Char, UInt8}, ::Union{Nothing, Char, UInt8, String}, ::Union{Char, UInt8}, ::Union{Nothing, Vector{String}}, ::Union{Nothing, Vector{String}}, ::Union{Nothing, String, Parsers.Format, Dates.DateFormat}, ::Any, ::Any, ::Any, ::Any, ::Any) at ~/.julia/packages/Parsers/a3jNK/src/Parsers.jl:81
  Parsers.Options(::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any, ::Any) at ~/.julia/packages/Parsers/a3jNK/src/Parsers.jl:59
  Parsers.Options(::Vector{String}, ::Union{Missing, Nothing, Vector{Tuple{Ptr{UInt8}, Int64}}}, ::Bool, ::Bool, ::UInt8, ::UInt8, ::Bool, ::UInt8, ::UInt8, ::UInt8, ::Union{Nothing, Tuple{Ptr{UInt8}, Int64}, UInt8}, ::UInt8, ::Union{Nothing, Vector{Tuple{Ptr{UInt8}, Int64}}}, ::Union{Nothing, Vector{Tuple{Ptr{UInt8}, Int64}}}, ::Union{Nothing, Parsers.Format}, ::Union{Nothing, Tuple{Ptr{UInt8}, Int64}}) at ~/.julia/packages/Parsers/a3jNK/src/Parsers.jl:59
Stacktrace:
  [1] CSV.Context(source::CSV.Arg, header::CSV.Arg, normalizenames::CSV.Arg, datarow::CSV.Arg, skipto::CSV.Arg, footerskip::CSV.Arg, transpose::CSV.Arg, comment::CSV.Arg, ignoreemptyrows::CSV.Arg, ignoreemptylines::CSV.Arg, select::CSV.Arg, drop::CSV.Arg, limit::CSV.Arg, buffer_in_memory::CSV.Arg, threaded::CSV.Arg, ntasks::CSV.Arg, tasks::CSV.Arg, rows_to_check::CSV.Arg, lines_to_check::CSV.Arg, missingstrings::CSV.Arg, missingstring::CSV.Arg, delim::CSV.Arg, ignorerepeated::CSV.Arg, quoted::CSV.Arg, quotechar::CSV.Arg, openquotechar::CSV.Arg, closequotechar::CSV.Arg, escapechar::CSV.Arg, dateformat::CSV.Arg, dateformats::CSV.Arg, decimal::CSV.Arg, truestrings::CSV.Arg, falsestrings::CSV.Arg, stripwhitespace::CSV.Arg, type::CSV.Arg, types::CSV.Arg, typemap::CSV.Arg, pool::CSV.Arg, downcast::CSV.Arg, lazystrings::CSV.Arg, stringtype::CSV.Arg, strict::CSV.Arg, silencewarnings::CSV.Arg, maxwarnings::CSV.Arg, debug::CSV.Arg, parsingdebug::CSV.Arg, validate::CSV.Arg, streaming::CSV.Arg)
    @ CSV ~/.julia/packages/CSV/VjW8G/src/context.jl:366
  [2] CSV.Context(source::String; header::Int64, normalizenames::Bool, datarow::Int64, skipto::Int64, footerskip::Int64, transpose::Bool, comment::Nothing, ignoreemptyrows::Bool, ignoreemptylines::Nothing, select::Nothing, drop::Nothing, limit::Nothing, buffer_in_memory::Bool, threaded::Nothing, ntasks::Nothing, tasks::Nothing, rows_to_check::Int64, lines_to_check::Nothing, missingstrings::Vector{String}, missingstring::String, delim::Nothing, ignorerepeated::Bool, quoted::Bool, quotechar::Char, openquotechar::Nothing, closequotechar::Nothing, escapechar::Char, dateformat::Nothing, dateformats::Nothing, decimal::UInt8, truestrings::Vector{String}, falsestrings::Vector{String}, stripwhitespace::Bool, type::Nothing, types::Nothing, typemap::Dict{Type, Type}, pool::Tuple{Float64, Int64}, downcast::Bool, lazystrings::Bool, stringtype::Type{InlineStrings.InlineString}, strict::Bool, silencewarnings::Bool, maxwarnings::Int64, debug::Bool, parsingdebug::Bool, validate::Bool)
    @ CSV ~/.julia/packages/CSV/VjW8G/src/context.jl:178
  [3] CSV.Context(source::String)
    @ CSV ~/.julia/packages/CSV/VjW8G/src/context.jl:178
  [4] _precompile_()
    @ CSV ~/.julia/packages/CSV/VjW8G/src/precompile.jl:7
  [5] top-level scope
    @ ~/.julia/packages/CSV/VjW8G/src/CSV.jl:95
  [6] include
    @ ./Base.jl:418 [inlined]
  [7] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::Nothing)
    @ Base ./loading.jl:1318
  [8] top-level scope
    @ none:1
  [9] eval
    @ ./boot.jl:373 [inlined]
 [10] eval(x::Expr)
    @ Base.MainInclude ./client.jl:453
 [11] top-level scope
    @ none:1
in expression starting at /home/admin/.julia/packages/CSV/VjW8G/src/CSV.jl:1
ERROR: Failed to precompile CSV [336ed68f-0bac-5ca0-87d4-7b16caf5d00b] to /home/admin/.julia/compiled/v1.7/CSV/jl_DNy0lQ.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IO, internal_stdout::IO, ignore_loaded_modules::Bool)
   @ Base ./loading.jl:1466
 [3] compilecache(pkg::Base.PkgId, path::String)
   @ Base ./loading.jl:1410
 [4] _require(pkg::Base.PkgId)
   @ Base ./loading.jl:1120
 [5] require(uuidkey::Base.PkgId)
   @ Base ./loading.jl:1013
 [6] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:997
```

</details>

**Comment by adigitoleo** [comment]
Also no problems when building from master.

**Comment by adigitoleo** [comment]
Woops, my issue was actually #965

**Comment by ufechner7** [comment]
I had the same problem on Rasperry Pi 4 (64bit Raspian), but it did disappear after running ]up .

That is strange... Why do I have to run ]up on a fresh installation to make things work?

**Comment by adigitoleo** [comment]
@ufechner7 Yes, I think this is what caused it for me as well, doing just `]add` straight away pulled an older version of the package. Maybe Pkg doesn't pull the latest registry information during install?

**Comment by Neuromancer43** [comment]
I have the same issue. It should have been closed since the [new release v0.10.2](https://github.com/JuliaData/CSV.jl/releases/tag/v0.10.2)already supports precompile in Julia v1.7.1. 

But the release can't be accessed in `Pkg.add`. So you can just add it using Github link.
`add https://github.com/JuliaData/CSV.jl/releases/tag/v0.10.2`

**Comment by jlperla** [comment]
Same problem on 100% clean environment on Julia 1.6.5 (and julia 1.7.2) on Windows, nothing in the top-level environment at all.

Adding in
```
[compat]
CSV = "=0.10.1"
```
to my project file makes things function perfectly well after it is instantiated.

Alternatively, I added in 
```
[compat]
Parsers = ">=2.2.2"
```
and it also worked.  Is the issue that https://github.com/JuliaData/CSV.jl/blob/main/Project.toml#L23  needs to lowerbound `2.2.2`?
